third-party/rsutils/include/rsutils/ios/field.h

```diff
@@ -0,0 +1,64 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+#pragma once
+
+#include <string>
+#include <iosfwd>
+
+
+namespace rsutils {
+namespace ios {
+
+
+// Many times when we write the contents of some structure to a stream, multiple "fields" are involved. We want to allow
+// a standard way of providing:
+//     - field separation
+//     - value separation
+//     - both same-line and indented (multiline) options for big dumps
+//     - flexibility: not requiring a field name, value, nor any formatting or either
+// 
+// Indented output puts each field on its own line. To enable multiline output, simply set the indent:
+//      os << rsutils::ios::indent();
+//      os << my_structure;
+//      os << rsutils::ios::unindent();
+// See group() for another way of doing this.
+//
+struct field
+{
+    // Separate one field from the previous:
+    //     os << field::separator << "whatever";
+    // If any indent is set, this will be on a separate line!
+    static std::ostream & separator( std::ostream & );
```

Comment:
The separator/value notation is not that clear. Maybe use key/value or name/value notation